src/UploadHandler.cpp

```diff
@@ -0,0 +1,180 @@
+
+	if (std::find(harmfulExtensions.begin(), harmfulExtensions.end(), extension) != harmfulExtensions.end())
+	{
+		return true;
+	}
+	return false;
+}
+
+// Other possible codes:
+// 413 Payload Too Large, 401 Unauthorized, 403 Forbidden, 500 Internal Server Error
+void UploadHandler::checkFiles(const HTTPRequest &request)
+{
+	std::vector<File> files = request.getFiles();
+	std::vector<File>::iterator it;
+
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+		if (it->headers.find("filename") == it->headers.end())
+		{
+			std::cout << "422 Unprocessable Entity (Error: file does not have a name)" << std::endl;
+			return;
+		}
+	}
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+		if (it->fileContent.empty())
+		{
+			std::cout << "422 Unprocessable Entity (Error: file is empty)" << std::endl;
+			return;
+		}
+	}
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+
+		std::string filename = it->headers.find("filename")->second;
+		std::string extension = filename.substr(filename.find_last_of(".") + 1);
+		if (extension.empty())
+		{
+			// TODO: have an html ?
+			std::cout << "415 Unsupported Media Type (Error: file has no extension)" << std::endl;
+			return;
+		}
+		if (isHarmfulExtension(extension))
+		{
+			std::cout << "415 Unsupported Media Type (Error: file has a harmful extension)" << std::endl;
+			return;
+		}
+	}
+}
+
+void createFile(HTTPRequest &request)
+{
+	const std::string uploadDir = "upload/";
+
+	std::vector<File> files = request.getFiles();
+	std::vector<File>::iterator it;
+
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+		std::string filePath = uploadDir + (it->headers.find("filename"))->second;
+
+		std::ofstream outfile(filePath.c_str());
+		if (outfile.is_open())
+		{
+			outfile << it->fileContent;
+			outfile.close();
+			std::cout << "File created successfully at " << filePath << std::endl;
+		}
+		else
+		{
+			std::cout << "422 Unprocessable Entity (Error creating a file at " << filePath << ")" << std::endl;
+		}
+	}
+}
+
+void UploadHandler::handleRequest(const HTTPRequest &request, HTTPResponse &response)
+{
+	checkFiles(request);
+	if (!request.getUploadBoundary().empty())
+	{
+		createFile(const_cast<HTTPRequest &>(request));
+		handleResponse(response, "success");
+	}
+	else
+	{
+		std::cout << "415 Unsupported Media Type" << std::endl;
+		handleResponse(response, "bad_request");
+	}
+}
+
+std::string readFileContents(const std::string &filePath)
+{
+	std::ifstream file(filePath.c_str());
+	if (!file)
+	{
+		std::cerr << "Could not open file: " << filePath << std::endl;
+		return "<html><body><h1>Internal Server Error</h1></body></html>";
+	}
+	std::stringstream buffer;
+	buffer << file.rdbuf();
+	return buffer.str();
+}
+
+void UploadHandler::handleResponse(HTTPResponse &response, const std::string &code)
+{
+	std::string fileContents;
+	int statusCode;
+	std::string statusDescription;
+
+	if (code == "success")
+	{
```

Comment:
I used an enum, I capitalize so it conforms to the way you created the enum in connection (but it seems unlike macros they are not capitalized)

Reply:
Why we didn't use a simple boolean here? Cause we have three values? 

Why are we using the status description? The response.setStatusCode works without description right? Or did someone change this?

Reply:
Indeed we use boolean because we have three possibilities, 2 are essential and we could reduce it to just that, but maybe we'll have more than 3 as well

response.setStatusCode takes a message (in main as well), I think @dantol29 implemented it, a while ago, and the goal of the message is internal (debugging)

Reply:
Regarding statusCode, statusDescription and response.setStatusCode we have 9 lines of code instead of 3. Why?

Is there any other way that we can't organize the code without the needs of macros, enums or "success' string for a simple if else condtion with three values?

Reply:
I changed it